src/rpc/server.cpp

```diff
@@ -474,13 +474,22 @@ static inline JSONRPCRequest transformNamedArguments(const JSONRPCRequest& in, c
 
 UniValue CRPCTable::execute(const JSONRPCRequest &request) const
 {
+    static const std::set<std::string> account_methods = {"getaccountaddress", "getaccount",
+        "getaddressesbyaccount", "getreceivedbyaccount", "listaccounts", "listreceivedbyaccount",
+        "setaccount", "move"};
+
     // Return immediately if in warmup
     {
         LOCK(cs_rpcWarmup);
         if (fRPCInWarmup)
             throw JSONRPCError(RPC_IN_WARMUP, rpcWarmupStatus);
     }
 
+    // Accounts RPC methods are deprecated
```

Comment:
This is a smaller change, however because `rpc/server.cpp` is supposed to not be specific to bitcoin, I'd prefer to put the deprecation check in the individual methods.

Reply:
I agree that it's more correct to have these in the individual methods, but see my rationale above:

> That's perhaps a little bit messy (rpc/server.cpp ideally wouldn't have knowledge of individual Bitcoin RPCs), but it saves a lot of code duplication and will be removed in V0.18, so I decided it was acceptable to have it there.

Are you strongly opposed to have this code in rpc/server.cpp, even if just temporarily? If so, I can change the PR to move this code to individual methods.

Reply:
> but it saves a lot of code duplication

Well it's just 8 deprecated methods. I'm fine either way.

Reply:
To be honest I prefer the duplication. If it's the about of work just say so and I'll provide a patch.

Reply:
>  If it's the about the amount of work

no, mostly amount of review. I'm happy to write the code if you're happy to review it :)